### ISSUE TITLE
Refresh data from a REST API

### ISSUE THREAD

**Issue by jamolina2021** [rrousselGit/river_pod#1163]
Hi,
If I have the following provider:
```
final dashboardInfo = FutureProvider<DashboardInfo>((ref) async {
  final dashboard = ref.watch(dashboardProvider);
  final dashboardInfo = await dashboard.getAllData();
  return dashboardInfo;
});

  Widget build(BuildContext context) {
    return ref.watch(dashboardInfo).when(
        loading: () => Container(
...
...
...
```
and I want to refresh the data from the REST API server, although no change has occurred on the client, what should be done? 

I have searched for all the methods that the riverpod providers incorporate and I have only found "refresh", but "refresh" resets the values, I just want the recalculation and notification to occur. 

Is riverpod ready to do this? 

Thanks for your help.
Regards,
Jose

**Comment by rrousselGit** [comment]
> I have searched for all the methods that the riverpod providers incorporate and I have only found "refresh", but "refresh" resets the values, I just want the recalculation and notification to occur

What exactly does this mean?

refresh should be the solution

**Comment by jamolina2021** [comment]
But refresh initializes the values, for example, if the provider creates a class with 2 variables, I initially initialize them to some values, if I then change the values and then do a refresh, the class is recreated and the modified values are lost.

**Comment by rrousselGit** [comment]
What else do you expect exactly?

**Comment by jamolina2021** [comment]
I just wait for the recalculation to be done, that is, the API call (dashboard.getAllData()) and the notification to occur.

**Comment by rrousselGit** [comment]
What exactly does that mean?

**Comment by jamolina2021** [comment]
That is, I want this code 

```
final dashboardProvider = Provider((ref) => Dashboard());

final dashboardInfo = FutureProvider<DashboardInfo>((ref) async {
  final dashboard = ref.watch(dashboardProvider);
  final dashboardInfo = await dashboard.getAllData();
  return dashboardInfo;
});

```

to be executed without the "Dashboard()" object being recreated, I simply want the code inside

```
  final dashboard = ref.watch(dashboardProvider);
  final dashboardInfo = await dashboard.getAllData();
  return dashboardInfo;
```

to be executed and the notification to refresh the UI is made. 

If I do
`ref.refresh(dashboardProvider);`
the object of the Dashboard class is recreated and all the configuration of that object becomes the initial configuration of the creation of an object of type Dashboard. 

When working with APIs, it is very normal for applications to include a refresh mechanism from server data, but it must respect all the class variables that may exist at that moment, since these variables are the ones that bring the data from the server one way or another. 

Sorry if I don't write correctly in English, I know very little English. :(

How would you do this "refresh" mechanism without losing the content of the objects?

Thank you very much for your help and your explanations.

**Comment by rrousselGit** [comment]
Why refresh `dashboardProvider`? Why not refresh `dashboardInfo` directly? 

`ref.refresh(dashboardInfo)`

**Comment by rrousselGit** [comment]
I don't get your question.

refresh recreates the provider state. In your case you didn't want to recreate a certain class, so you shouldn't refresh it. Instead you wanted to refresh something else

**Comment by jamolina2021** [comment]
Exactly, I just want to recreate the information that a certain object has from an API, but I don't want to recreate the object itself.
As I have to continue with some more complex developments than this, if I find an example that involves directly using a provider to refresh the UI that has a class that receives data from an API, I will tell you about it. But it is very possible that using the technique that you have told me, a "refresh" can be done without using the original class that has the data from the API. 
You are a crack. :)
Thank you very much for the explanation and for your wonderful library. 🥇

**Comment by rrousselGit** [comment]
Glad to help
